demos/src/iot_demo_mqtt.c

```diff
@@ -417,6 +418,21 @@ static int _establishMqttConnection( bool awsIotMqttMode,
     connectInfo.keepAliveSeconds = KEEP_ALIVE_SECONDS;
     connectInfo.pWillInfo = &willInfo;
 
+    if( pCredentials != NULL )
+    {
+        connectInfo.pUserName = pCredentials->pUserName;
+        connectInfo.userNameLength = ( uint16_t ) pCredentials->userNameSize;
+        connectInfo.pPassword = pCredentials->pPassword;
+        connectInfo.passwordLength = ( uint16_t ) pCredentials->passwordSize;
+    }
+    else
+    {
+        connectInfo.pUserName = NULL;
```

Comment:
The default initializer should set these all to 0, so this block isn't necessary.  May still be helpful as a demo though, so I understand if you want to keep it.